### PR TITLE
Allow users to config rspec retry command

### PR DESCRIPTION
**Pull request by amybiyuliu**
### Description
Currently test splitter supports a max retry limit and retry failed tests with the test command + `--only-failures`.
This PR allows user to config the retry command such that user can use other retry options (such as ` --fail-fast`, ` --next-failure`).

### Changes
Add new optional env variable `BUILDKITE_SPLITTER_RETRY_CMD`
Run failed tests with the `BUILDKITE_SPLITTER_RETRY_CMD` if user provided the env, otherwise run failed tests with the default retry command (test command + `--only-failures`)

### Testing

Testing against bk/bk local with retry command `bundle exec rspec --only-failures --fail-fast`

Two failure cases after first run Failed examples:

rspec ./spec/models/changelog/publisher_spec.rb:18 # Changelog::Publisher false2
rspec ./spec/models/changelog/publisher_spec.rb:17 # Changelog::Publisher false1

Retry stop running the remaining failed cases from the last run after seeing the first test failure
Screenshot 2024-07-11 at 2 51 18 pm

### CODE REVIEW

File: internal/config/env.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func (c Config) DumpEnv() map[string]string {
 		"BUILDKITE_PARALLEL_JOB",
 		"BUILDKITE_SPLITTER_DEBUG_ENABLED",
 		"BUILDKITE_SPLITTER_RETRY_COUNT",
+		"BUILDKITE_SPLITTER_RETRY_CMD",
```
Comment (amybiyuliu) [inline internal/config/env.go:46]: will need to add this env to the whitelist when persisting the test plan

File: README.md [hunk 2]
```diff
@@ -44,7 +44,8 @@ The following environment variables can be used optionally to configure your Tes
...
 | ---- | ---- | ----------- |
 | `BUILDKITE_SPLITTER_DEBUG_ENABLED` | `false` | Flag to enable more verbose logging. |
-| `BUILDKITE_SPLITTER_RETRY_COUNT` | `0` | The number of retries permitted. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries only the failing tests for a maximum of `BUILDKITE_SPLITTER_RETRY_COUNT` times. For Rspec, the Test Splitter runs `BUILDKITE_SPLITTER_TEST_CMD` with `--only-failures` as the retry command. |
+| `BUILDKITE_SPLITTER_RETRY_COUNT` | `0` | The number of retries permitted. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries only the failed tests for a maximum of `BUILDKITE_SPLITTER_RETRY_COUNT` times with the retry command `BUILDKITE_SPLITTER_RETRY_CMD`. |
+| `BUILDKITE_SPLITTER_RETRY_CMD` | `BUILDKITE_SPLITTER_TEST_CMD` with `--only-failures` | The command to rerun the failed tests. For Rspec, it depends on `example_status_persistence_file_path` to be defined as part of your Rspec configuration for the retry to work. Please see [Rspec documentation](https://rspec.info/features/3-12/rspec-core/command-line/only-failures/) for available retry options.
```
Comment (nprizal) [inline README.md:48]: Can we sort this alphabetically?

**Comment by amybiyuliu** [comment]
The PR for adding the new env to the env white list: https://github.com/buildkite/buildkite/pull/17829

File: README.md [hunk 3]
```diff
@@ -44,7 +44,8 @@ The following environment variables can be used optionally to configure your Tes
...
 | ---- | ---- | ----------- |
 | `BUILDKITE_SPLITTER_DEBUG_ENABLED` | `false` | Flag to enable more verbose logging. |
-| `BUILDKITE_SPLITTER_RETRY_COUNT` | `0` | The number of retries permitted. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries only the failing tests for a maximum of `BUILDKITE_SPLITTER_RETRY_COUNT` times. For Rspec, the Test Splitter runs `BUILDKITE_SPLITTER_TEST_CMD` with `--only-failures` as the retry command. |
+| `BUILDKITE_SPLITTER_RETRY_CMD` | `BUILDKITE_SPLITTER_TEST_CMD` with `--only-failures` | The command to rerun the failed tests. For Rspec, it depends on `example_status_persistence_file_path` to be defined as part of your Rspec configuration for the retry to work. Please see [Rspec documentation](https://rspec.info/features/3-12/rspec-core/command-line/only-failures/) for available retry options.
+| `BUILDKITE_SPLITTER_RETRY_COUNT` | `0` | The number of retries permitted. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries only the failed tests for a maximum of `BUILDKITE_SPLITTER_RETRY_COUNT` times with the retry command `BUILDKITE_SPLITTER_RETRY_CMD`. |
```
Comment (wooly) [inline README.md:48]:
```suggestion
| `BUILDKITE_SPLITTER_RETRY_COUNT` | `0` | The number of retries performed. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries only the failed tests for a maximum of `BUILDKITE_SPLITTER_RETRY_COUNT` times with the retry command `BUILDKITE_SPLITTER_RETRY_CMD`. |
```

**Comment by wooly** [comment]
Happy for this to go in pending some integration testing.

**Comment by wooly** [comment]
@amybiyuliu @nprizal and I integration tested this together and we are happy that it is working as expected, so will give it the ✅

**Review by wooly — APPROVED** [review]
Just the final nit on the README description.

**Comment by wooly** [comment]
Linear ticket is here: https://linear.app/buildkite/issue/TAT-150/allow-users-to-config-rspec-retry-command-retry-step-2

**Comment by wooly** [comment]
Fixes TAT-150

**Pull request merged**